test/unittest/blas3/blas3_gemm_tall_skinny_test.cpp

```diff
@@ -0,0 +1,105 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename blas3_gemm_tall_skinny_test.cpp
+ *
+ **************************************************************************/
+
+#include "blas_test.hpp"
+
+template <typename scalar_t>
+using combination_t =
+    std::tuple<int, int, int, char, char, scalar_t, scalar_t, int, int, int>;
+
+const auto combi = ::testing::Combine(::testing::Values(7, 65),      // m
```

Comment:
Do these test values hit all the possible code paths in the TSGEMM?

Reply:
Not exactly, I will explain why, and why it's not a problem:

- heuristics are used independently on each platform to use different tile dimensions. Having the tests hit all the code paths, and all the possible parameterizations for each platform would require a number of tests that we cannot afford if we want to keep the testing reasonably short. I have hit all these parameterizations with verified benchmarks when optimizing the heuristics, though.
- Also, the tests only use a ldc value different than m, which is handled in a different branch in the executor. In the benchmarks the ldc value is always m so this branch is verified by the benchmarks. I realize this can be seen as an issue and I can change the test cases, but it will double their number.

There's a trade-off here, between exhaustivity and testing time.

Reply:
I added the case where `ldc_mul` = 1 to have more exhaustive testing as the test time is still reasonable (17s).